dashboard/test/lib/datastore_cache_test.rb

```diff
@@ -23,10 +72,22 @@ def setup
     @data_adapter.set(key, 'not the real value')
     cache = DatastoreCache.new @data_adapter, cache_expiration: 0.01
 
+    listener = FakeListener.new
+    cache.add_change_listener(listener)
+
     @data_adapter.set(key, value)
     sleep 0.1
 
     assert_equal value, cache.get(key)
+    assert listener.changed, 'on_change should be called after polled updated'
+    listener.reset
+    sleep 0.1
+    assert !listener.changed, 'on_change should not be called when polling detects no changes'
+
+    @data_adapter.set(key, 'new value')
+    sleep 0.1
+    assert_equal 'new value', cache.get(key)
+    assert listener.changed, 'on_change should be called after second polled updated'
```

Comment:
I just tried timecop, unfortunately it can't be used here: moving forward in time does not cause sleeps that are already in progress to finish early, so it's not possible to use Timecop to ensure that the polling has occurred.